### PR TITLE
quickstart: recommend PyPA build instead of pep517.build

### PR DESCRIPTION
**Pull request by hexagonrecursion**
## Summary of changes

Docs update to recommend PyPA build instead of pep517.build (see https://github.com/pypa/pep517/issues/91). Similar to #2504 but for docs/userguide/quickstart.rst

### CODE REVIEW

**Comment by hexagonrecursion** [comment]
I think this is too minor to deserve a changelog entry.

File: docs/userguide/quickstart.rst [hunk 1]
```diff
@@ -59,11 +59,11 @@ This is what your project would look like::
         setup.cfg
         mypackage/__init__.py
 
-Then, you need an installer, such as `pep517 <https://pypi.org/project/pep517/>`_
-which you can obtain via ``pip install pep517``. After downloading it, invoke
+Then, you need an installer, such as `PyPA build <https://pypa-build.readthedocs.io/en/latest/>`_
```
Comment (webknjaz) [inline docs/userguide/quickstart.rst:62]: I like the idea of linking the python-build docs but could you please add their RTD page to intersphinx in the `conf.py` instead?

Comment (hexagonrecursion) [reply docs/userguide/quickstart.rst]: I am new to sphinx. I have never heard of conf.py or intersphinx. I am not familiar with the nuances of rst syntax. I don't know what a domain is. I can't find the documentation for the `std` domain anywhere. I have had to read a whole bunch of documentation to figure out what you want me to do))))))

done

**Comment by hexagonrecursion** [comment]
Nope. Not done. Now I get:
```
Warning, treated as error:
/home/user/git/docs/userguide/quickstart.rst:62:Mismatch: both interpreted text role prefix and reference suffix.
ERROR: InvocationError for command /home/user/git/.tox/docs/bin/python -m sphinx -W . /home/user/git/build/html (exited with code 2)
```
and I have no idea what this means and how to fix this. Searching did not yield anything helpful either yet.

**Comment by hexagonrecursion** [comment]
30 minutes: how log it took me to figure out that I should remove the _. The error message could use improvement.

**Comment by hexagonrecursion** [comment]
Sorry for acting unprofessionally. I have no idea why I'm in a bad mood today.

File: docs/userguide/quickstart.rst (hunk 1 above)
Comment (webknjaz) [reply docs/userguide/quickstart.rst]: Oh, it's the same as just `:doc:`, but intersphinx still exposes it as `:std:doc:`, I guess to produce less conflicts...

Thanks for approaching this issue, anyway :)

File: docs/conf.py [hunk 2]
```diff
@@ -1,4 +1,10 @@
-extensions = ['sphinx.ext.autodoc', 'jaraco.packaging.sphinx', 'rst.linker']
+extensions = [
+    'sphinx.ext.autodoc',
+    'jaraco.packaging.sphinx',
+    'rst.linker',
+    'sphinx.ext.intersphinx',
+    'sphinx.ext.extlinks',
+]
```
Comment (webknjaz) [inline docs/conf.py:7]: Please revert this change, the top part of this module is merged-in from the shared skeleton, and changing it will mess things up for the maintainers.

File: docs/userguide/quickstart.rst (hunk 1 above)
Comment (webknjaz) [reply docs/userguide/quickstart.rst]: (sorry for the typo in my suggestion)

File: docs/conf.py (hunk 2 above)
Comment (hexagonrecursion) [reply docs/conf.py:7]: There should be a comment in conf.py warning contributors about the skeleton

Comment (hexagonrecursion) [reply docs/conf.py]: done

Comment (webknjaz) [reply docs/conf.py]: Yeah, probably. But it'd need to go to https://github.com/jaraco/skeleton first. This is something @jaraco may want to address.

**Review by webknjaz — APPROVED** [review]
LGTM :heart:

**Review by jaraco — APPROVED** [review]

**Comment by jaraco** [comment]
Thanks for the PR and the thoughtful review.
